data/sensors_recent.php

```diff
@@ -292,18 +400,36 @@ function output_cell($rowspan, $data) {
 			} else {
 				
 				$distance = false;
+				#printf("432 row[latitude]=%s gwdata[latitude]=%s<br>", $row['latitude'], $gwdata['latitude']);
 				if ($row['latitude'] && array_key_exists('latitude', $gwdata) && $gwdata['latitude']) {
 					$distance = haversineGreatCircleDistance($row['latitude'], $row['longitude'], $gwdata['latitude'], $gwdata['longitude']);
 				}
+
 				$gw_id = $gwdata['gtw_id'];
-				echo("  <td>" . gateway_button($gw_id, $gwdata) . "</td>\n");
+				if ($gw_id == "unknown") {
+					echo("  <td><I>" . $gw_id . "</i></td>\n");
+				} else if (($gwdata['longitude'] == 0) && ($gwdata['latitude'] == 0)) {
+					echo("  <td>" . $gw_id . "</td>\n");
```

Comment:
Je roept hier nu `gateway_button` niet aan als er geen locatie is voor de gateway, maar dat betekent dat je dan ook niet op gateway id kunt filteren. Daarnaast checkt `gateway_button` zelf al of er een locatie is of niet, dus volgens mij kunnen deze twee regels weg.